runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/options/DataflowPipelineOptions.java

```diff
@@ -82,6 +82,14 @@
   void setUpdate(boolean value);
 
   /**
+   * Whether the runner should simply generate a template and exit.
+   */
+  @Description("Whether the runner should simply generate a template and exit.")
+  // Default
+  boolean getTemplateRunner();
+  void setTemplateRunner(boolean value);
```

Comment:
It looks like you are proposing that a user who wants to create a template should set both `--dataflowJobFile` and `--templateRunner`.

I don't think this is the right API:
* They can set `--templateRunner` and that will have no effect nor print a warning? Seems odd.
* `--dataflowJobFile` seems misleading as a name.

Thinking about how this might work more simply:

* Keep `--dataflowJobFile` meaning what it currently means, and factor the logic that saves a job graph so that it's reusable?
* Add a new `--templateLocation` argument (verify it's a GS path) and have that both turn on templating and write the job graph to the right place.

Additional Qs:
* Where do you want the staged jars to live? Might make sense if they live under a subfolder of the `templateLocation` rather than in the `stagingLocation`. In fact, probably, `stagingLocation` should be a RVP for these scenarios.